lib/requestResponseLink.ts

```diff
@@ -205,13 +205,13 @@ export class RequestResponseLink implements ReqResLink {
    */
   static async create(connection: Connection, senderOptions: SenderOptions, receiverOptions: ReceiverOptions): Promise<RequestResponseLink> {
     if (!connection) {
-      throw new Error(`Please provide a connection to create the sender/receiver link on the same session.`);
+      throw new TypeError(`Please provide a connection to create the sender/receiver link on the same session.`);
     }
     if (!senderOptions) {
-      throw new Error(`Please provide sender options.`);
+      throw new TypeError(`Please provide sender options.`);
     }
     if (!receiverOptions) {
-      throw new Error(`Please provide receiver options.`);
+      throw new TypeError(`Please provide receiver options.`);
```

Comment:
Lets remove the above 3 checks as well.

Missing connection will throw errors a few lines below when creating session.
Senders and Receivers can be created without the options as well. `createSender` and `createReceiver` functions take optional parameter